lib/src/rules/avoid_function_literals_in_foreach_calls.dart

```diff
@@ -54,6 +54,25 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitMethodInvocation(MethodInvocation node) {
+    var exp = node.target;
+    var chainCount = 0;
+    while (exp is PrefixedIdentifier ||
+        exp is MethodInvocation ||
+        exp is PropertyAccess) {
+      if (exp is PrefixedIdentifier) {
```

Comment:
In trying to convince myself that this is right I found myself wanting a bit of text in the lint description to describe the exception.

That said, @scheglov, @bwilkerson: is there a better way to do this?

Reply:
A better way to do what, specifically? There isn't any utility method that will count the number of method invocations in a chain, if that's what you're asking.

Reply:
Yep.  That's what I was trying to ask (but without ☕️).

My biggest remaining question is whether the hole should be poked open for field/getter accesses (as implemented) vs. only for chained function calls (as I thought).

Any opinion?

Reply:
I thinks is addressed in the other conversations